src/operators/ai.onnx/ConvTranspose/11/execute_operator__ai_onnx__convtranspose__11__T_tensor_float.c

```diff
@@ -0,0 +1,193 @@
+    // TRACE_VAR(2, true, group, "%" PRId64);
+    // TRACE_ARRAY(2, true, kernel_shape, , n_kernel_shape, "%" PRId64);
+    // TRACE_ARRAY(2, true, output_padding, , n_output_padding, "%" PRId64);
+    // TRACE_ARRAY(2, true, output_shape, , n_output_shape, "%" PRId64);
+    // TRACE_ARRAY(2, true, pads, , n_pads, "%" PRId64);
+    // TRACE_ARRAY(2, true, strides, , n_strides, "%" PRId64);
+
+    Onnx__TensorProto *o_Y = searchOutputByName(ctx, 0);
+    printf("%p\n", o_Y->float_data);
+
+    // TRACE_TENSOR(2, true, o_Y);
+
+    /* DO CALCULATION HERE */
+
+    const int inputSizeX = i_X->dims[3];
+    const int inputSizeY = i_X->dims[2];
+    const int inputChannels = i_X->dims[1];
+
+    const float *input = i_X->float_data;
+    const float *weights = i_W->float_data;
+
+    const int kernelSizeX = i_W->dims[3];
+    const int kernelSizeY = i_W->dims[2];
+    const int outputChannels = i_W->dims[1];
+
+    int strideX = 1;
+    int strideY = 1;
+
+    if(a_strides != NULL) {
+        strideX = a_strides->ints[1];
+        strideY = a_strides->ints[0];
+    }
+
+    int dilationsX = 1;
+    int dilationsY = 1;
+
+    if(a_dilations != NULL) {
+        dilationsX = a_dilations->ints[1];
+        dilationsY = a_dilations->ints[0];
+    }
+
+    int padStartY = 0;
+    int padStartX = 0;
+    int padEndY = 0;
+    int padEndX = 0;
+
+    if(a_pads != NULL) {
+        assert(a_pads->n_ints == 4);
+
+        padStartY = a_pads->ints[0];
+        padStartX = a_pads->ints[1];
+        padEndY = a_pads->ints[2];
+        padEndX = a_pads->ints[3];
+    }
+
+    const int outputSizeX = calcOutputSize(inputSizeX, kernelSizeX, strideX, dilationsX, 0, 0);
+    const int outputSizeY = calcOutputSize(inputSizeY, kernelSizeY, strideY, dilationsY, 0, 0);
+
+    const int outputSizePaddedX = calcOutputSize(inputSizeX, kernelSizeX, strideX, dilationsX, padStartX, padEndX);
+    const int outputSizePaddedY = calcOutputSize(inputSizeY, kernelSizeY, strideY, dilationsY, padStartY, padEndY);
+
+    //if badding is needed we need an extra tensor and applay padding later
+    float* output = NULL;
+    if(outputSizeX * outputSizeY != outputSizePaddedX * outputSizePaddedY) {
+        output = (float*)malloc(outputSizeX * outputSizeY * outputChannels * sizeof(float));
+    } else {
+        output = o_Y->float_data;
+    }
+
+    //fill with bias
+    if(i_B == NULL) {
+        for(int c=0; c<outputChannels; c++) {
+            for(int y=0; y<outputSizeY; y++) {
+                for(int x=0; x<outputSizeX; x++) {
+                    output[calcArrayPos3D(x,y,c,outputSizeX, outputSizeY)] = 0;
+                }
+            }
+        }
+    } else {
+        for(int c=0; c<outputChannels; c++) {
+            for(int y=0; y<outputSizeY; y++) {
+                for(int x=0; x<outputSizeX; x++) {
+                    output[calcArrayPos3D(x,y,c,outputSizeX, outputSizeY)] = i_B->float_data[c];
+                }
+            }
+        }
+    }
```

Comment:
why do this here and not in the actual convolution?

Reply:
Because the convolution is a sum calculated with += in a loop and we need a starting value which is either 0 or the bias value

Reply:
thanks, I forgot you loop for each channel over the same output cell m(
would still modify this to sth like
```c
        for(int c=0; c<outputChannels; c++) {
            float bias = i_B?i_B->float_data[c]:0;
            for(int y=0; y<outputSizeY; y++) {
                for(int x=0; x<outputSizeX; x++) {
                    output[calcArrayPos3D(x,y,c,outputSizeX, outputSizeY)] = bias;
                }
            }
        }
    }

```

Reply:
I didn't want if statements inside the loop but yes its only one if per channel so I can change it

Reply:
why no if statements inside the loop? sounds like you are trying to solve performance issues (did you have any?)

Reply:
I didn't had performance issues but neither did I want to make the program unnecessary slower. In this case here it doesn't matter.